### ISSUE TITLE
generate black image

### ISSUE THREAD

**Issue by yanre-hyd** [garibida/cross-image-attention#13]
thanks for your job! i found a error in the code,whatever image i use which always generated black image,how to solute this error?looking forward to your answer!

**Comment by Mofr77** [comment]
> thanks for your job! i found a error in the code,whatever image i use which always generated black image,how to solute this error?looking forward to your answer!

I also encountered the same problem, have you solved it?

**Comment by yanre-hyd** [comment]
> > 谢谢你的工作！我在代码中发现了一个错误，无论我使用什么图像，它总是生成黑色图像，如何解决这个错误？期待您的答复！
> 
> 我也遇到了同样的问题，你解决了吗？

yes! i found a part of code is strange which is in adain.py, in the calc_mean_std function,there is a "mask" for formal parameter,it is none by default,so it will generate black image,so we need delete a part of codes , it is"if mask is not None:..." reserve  the codes behind the "else",although it not generate black image,but it have not good effect,so i am also finding the reason now

**Comment by Mofr77** [comment]
> > > 谢谢你的工作！我在代码中发现了一个错误，无论我使用什么图像，它总是生成黑色图像，如何解决这个错误？期待您的答复！
> > 
> > 
> > 我也遇到了同样的问题，你解决了吗？
> 
> yes! i found a part of code is strange which is in adain.py, in the calc_mean_std function,there is a "mask" for formal parameter,it is none by default,so it will generate black image,so we need delete a part of codes , it is"if mask is not None:..." reserve the codes behind the "else",although it not generate black image,but it have not good effect,so i am also finding the reason now

Thank you. It no longer generates black images, but the results are not satisfactory. I will continue to find the reason. Thank you again❤❤.

**Comment by yuxu915** [comment]
@1289600760 hi, really thanks for you suggestion, and have you finally find the reason?

**Comment by yanre-hyd** [comment]
> @1289600760你好，非常感谢你的建议，你最终找到原因了吗？

sorry ，i didn't found the reason

**Comment by yanre-hyd** [comment]
> @1289600760你好，非常感谢你的建议，你最终找到原因了吗？

 it have not good effect,so this method is not adapt my dataset,

**Comment by tianyang-wang35** [comment]
@1289600760 Could we please communicate with each other? if you not mind, please add me:13186212510, very thanks to you.

**Comment by yuval-alaluf** [comment]
If you're getting a black image, this could be because the prompt you are passing the model does not match the image that you are giving the model. This then causes the automatic segmentation that we use for AdaIN to not segment the object correctly. 
There are two ways to fix this: 
1. Refine the prompt you are using and ensure that the prompt matches the object in the image.
2. Disable the use of the AdaIN using the appropriate flag in the config.
Hope this helps

**Comment by niqbal996** [comment]
Hi @yuval-alaluf , 

I met the same issue and your 1st solution mentioned above does generate a reasonable image however, my results do not seem that good. I am generating this on a plant based image with domain names like "sugarbeets, crops, weeds, agriculture" and with **--use_masked_adain True** I still get black images.  What to do if I want to use automatic segmentation method with AdaIN? Do you have any suggestions for transfering this on my custom domain dataset?
